BEIMA.Backend/DeviceFunctions/AddDevice.cs

```diff
@@ -16,45 +17,80 @@ namespace BEIMA.Backend.DeviceFunctions
     /// <summary>
     /// Handles a request to add a single device.
     /// </summary>
-    public static class AddDevice
+    public class AddDevice
     {
+        private readonly IStorageProvider _storage;
+
+        /// <summary>
+        /// Constructor for the AddDevice Function
+        /// </summary>
+        /// <param name="storage">Depedency injected storage provider</param>
+        public AddDevice(IStorageProvider storage)
```

Comment:
So, I'm concerned that using this constructor is not going to play nicely with the current way we are using these endpoints. This seems to suggest that the request or the application itself should just know what storage provider to put in here on its own, which I don't think is the case. I think we would be better suited doing something like we are with the mongo connector, using instances from static or singleton classes.

Reply:
So it is singleton, at application startup the environment variables are read and a single storage provider is created just like the mongo connection. The only difference is that the application holds that reference and passes it to the endpoint rather then going to a class a getting the static instance held there.

Reply:
The mongo connector is also a singleton, can we be consistent and handle singletons the same way?

Reply:
I see how you have set up the unit tests to provide the storage provider, but I'm not seeing how when we launch the application on it's own, how it will figure out how to populate that storage provider? I think this could be why the FTs are failing as well.

Reply:
Startup.cs gets ran automatically on startup by the application which is how it gets created. 

The FT's are failing because I'm still in the process of getting the mulitpartform request created correctly. That was why I opened the pullrequest early so I could get a jumpstart on feedback while trying to debug it. The request gets sent fine, but req.ReadFormAsync returns null. Manually hitting the endpoints from postman works just fine, same when running the normal tests so I don't think the dependency injection is the issue.

Reply:
Ah ok, that makes sense. I do still agree with Ashlyn though in that I think it would be in our best interest to be consistent with how we are using the MongoConnector in the endpoints, do you think that it is possible to adjust it so that we don't need to use the constructor? Or perhaps even the other way around?

Reply:
Yeah that would be possible